lighthouse-core/audits/long-tasks.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('./audit.js');
+const NetworkRecords = require('../computed/network-records.js');
+const i18n = require('../lib/i18n/i18n.js');
+const MainThreadTasks = require('../computed/main-thread-tasks.js');
+const BootupTime = require('./bootup-time.js');
+
+const UIStrings = {
+  /** Title of a diagnostic LH audit that provides details on the longest running tasks that occur when the page loads. */
+  title: 'Long main thread tasks',
+  /** Description of a diagnostic LH audit that tells the user to minimize the amount of long-running tasks on a page. */
+  description: 'Lists the longest tasks on the main thread, ' +
+    'useful for identifying worst contributors to input delay.',
+  /** [ICU Syntax] Label for an audit that tells the user to minimize the amount of long-running tasks on a page. */
+  displayValue: `{itemCount, plural,
+  =1 {1 long task found}
+  other {# long tasks found}
+  }`,
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class LongTasks extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'long-tasks',
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['traces', 'devtoolsLogs'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const tasks = await MainThreadTasks.request(trace, context);
+    const devtoolsLog = artifacts.devtoolsLogs[LongTasks.DEFAULT_PASS];
+    const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+
+    const jsURLs = BootupTime.getJavaScriptURLs(networkRecords);
+    const longtasks = [...tasks]
+      .filter(t => t.duration >= 50 && !t.unbounded && !t.parent)
+      .sort((a, b) => b.duration - a.duration)
+      .slice(0, 20);
```

Comment:
startTime is going to be really tricky. the best way is going to be grabbing the pessimistic TTI graph and finding the startTime of the matching event id. 

That's a pain though, do we need to show it at first? It's not like the simulated start time is going to be all that meaningful for tracking it down in the trace either since it's a completely different execution path :/

Reply:
uhh why isn't it just multiplier * startTime?

Reply:
yup :)

not only that but some tasks might also depend on a series of network requests that do not scale according to the `multiplier`

Reply:
I'm ok with leaving out start time for now. We'll need it for a future pass, if we want to visualize as a timeline.